File: task_equivalents/llm_translations/gemini/base/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
College <- read.csv('data/csv/College.csv')
rownames(College) <- College[, 1]
College <- College[, -1]

College$Private01 <- ifelse(College$Private == 'Yes', 1, 0)
College$Private <- NULL

X <- College[, names(College) != 'Apps']
X[] <- lapply(X, as.numeric)
Y <- College$Apps

set.seed(1)
train_index <- sample(seq_len(nrow(X)), size = floor(nrow(X) * 0.5))
xtrain <- X[train_index, ]
xtest <- X[-train_index, ]
ytrain <- Y[train_index]
ytest <- Y[-train_index]

File: task_equivalents/llm_translations/gemini/base/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import pandas as pd
import numpy as np

College = pd.read_csv('data/csv/College.csv', index_col=0)

College['Private01'] = (College['Private'] == 'Yes').astype(int)
College = College.drop(columns=['Private'])

X = College.drop(columns=['Apps']).apply(pd.to_numeric)
Y = College['Apps']

np.random.seed(1)
train = np.random.choice(len(College), len(College) // 2, replace=False)
test = np.setdiff1d(np.arange(len(College)), train)

xtrain = X.iloc[train]
xtest  = X.iloc[test]
ytrain = Y.iloc[train]
ytest  = Y.iloc[test]